### PR TITLE
[_]: feat/handle lost disputes through the webhook

### PR DESCRIPTION
**Pull request by xabg2**
When a dispute is closed with status lost, if it is for a subscription, we will cancel it and if it is a lifetime, the lifetime status will be set to false. 

For both cases the storage will be changed to 1GB (free plan).

This PR also contains the update of a type that has become obsolete and the correction of a grammatical problem of a function.

### CODE REVIEW

File: src/webhooks/handleInvoicePaymentFailed.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default async function handleInvoicePaymentFailed(
   invoice: Stripe.Invoice,
   objectStorageService: ObjectStorageService,
   paymentService: PaymentService,
-  logger: FastifyLoggerInstance,
+  logger: FastifyBaseLogger,
```
Comment (sg-gs) [inline src/webhooks/handleInvoicePaymentFailed.ts:25]: Why this change is needed?

Comment (xabg2) [reply src/webhooks/handleInvoicePaymentFailed.ts:25]: The change is not needed for this PR, but FastifyLogerInstance is deprecated, so I updated it to the new one.

File: src/webhooks/handleDisputeLost.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  stripe: Stripe;
+  paymentService: PaymentService;
+  usersService: UsersService;
+  storageService: StorageService;
+  cacheService: CacheService;
+  log: FastifyBaseLogger;
+  config: AppConfig;
+}
+
+export async function handleDisputeLost({
```
Comment (sg-gs) [inline src/webhooks/handleDisputeLost.ts:21]: `handleDisputeResult` could be a better name, as you do not know if the dispute has been lost or not, for what I see in the below checks

File: src/webhooks/handleDisputeLost.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  paymentService,
+  usersService,
+  storageService,
+  cacheService,
+  log,
+  config,
+}: HandleDisputeLostProps) {
+  const isDisputeLost = charge.status === 'lost';
+  const chargeId = charge.charge as string;
+
```
Comment (sg-gs) [inline src/webhooks/handleDisputeLost.ts:33]: Return if the dispute is not lost, so you reduce the cognitive complexity of the if that wraps the entire `try-catch`

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/webhooks/handleInvoicePaymentFailed.ts (hunk 1 above)
Comment (sg-gs) [reply src/webhooks/handleInvoicePaymentFailed.ts:25]: Okay, add the reference link of the Fastify documentation that mentions the deprecation in the description of this PR so we keep track of that in case anything happens.
